splash/src/main/kotlin/edu/artic/splash/SplashActivity.kt

```diff
@@ -20,15 +24,38 @@ class SplashActivity : BaseViewModelActivity<SplashViewModel>() {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+
         viewModel.percentage
                 .map {
                     "Percentage : %.2f".format(it * 100)
                 }
                 .onErrorReturn {
-                    it.localizedMessage
+                    "Error: ${it.localizedMessage}"
+                }
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe {
+                    if (BuildConfig.DEBUG) {
+                        if (it.contains("Error")) {
+                            percentText.text = it
+                            percentText.visibility = View.VISIBLE
+                        } else {
+                            percentText.visibility = View.GONE
+                        }
+                    }
                 }
-                .bindToMain(percentText.text())
                 .disposedBy(disposeBag)
+
+        viewModel.percentage
+                .subscribe {
+                    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+                        progressBar.setProgress((it * 100).toInt(), true)
+                    } else {
+                        progressBar.progress = it.toInt()
```

Comment:
...not `(it * 100).toInt()`?

Reply:
My bad. Totally forgot for else case. Updated.